### PR TITLE
RHOAIENG-15875: fix(deps): upgrade protobuf package to unaffected version

### PR DESCRIPTION
**Pull request by jiridanek**
https://issues.redhat.com/browse/RHOAIENG-15875

## How Has This Been Tested?

Controllers start and seem to operate normally. This is low-risk dependency upgrade, so I'm happy to leave extensive testing to when we have Konflux build for this.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: components/notebook-controller/go.sum [hunk 1]
```diff
@@ -200,10 +198,8 @@ google.golang.org/protobuf v0.0.0-20200228230310-ab0ca4ff8a60/go.mod h1:cfTl7dwQ
 google.golang.org/protobuf v1.20.1-0.20200309200217-e05f789c0967/go.mod h1:A+miEFZTKqfCUM6K7xSMQL9OKL/b6hQv+e19PK+JZNE=
 google.golang.org/protobuf v1.21.0/go.mod h1:47Nbq4nVaFHyn7ilMalzfO3qCViNmqZ2kzikPIcrTAo=
 google.golang.org/protobuf v1.23.0/go.mod h1:EGpADcykh3NcUnDUJcl1+ZksZNG86OlYog2l/sGQquU=
-google.golang.org/protobuf v1.26.0-rc.1/go.mod h1:jlhhOSvTdKEhbULTjvd4ARK9grFBp09yW+WbY/TyQbw=
-google.golang.org/protobuf v1.26.0/go.mod h1:9q0QmTI4eRPtz6boOQmLYwt+qCgq0jsYwAQnmE0givc=
```
Comment (jstourac) [inline components/notebook-controller/go.sum:204]: Do we know why we have also this older version here? Are we sure we can remove it safely? :thinking:

Comment (jiridanek) [reply components/notebook-controller/go.sum:204]: I discussed in https://redhat-internal.slack.com/archives/C017W7H1J81/p1731656143799479, seems that go uses the go.mods to run minimal version selection

> The go.sum file may contain hashes for multiple versions of a module. The go command may need to load go.mod files from multiple versions of a dependency in order to perform [minimal version selection](https://go.dev/ref/mod#minimal-version-selection). go.sum may also contain hashes for module versions that aren’t needed anymore (for example, after an upgrade). go mod tidy will add missing hashes and will remove unnecessary hashes from go.sum.
https://go.dev/ref/mod#go-sum-files

**Pull request merged**